frontend/components/SetupPage/SelectYourAgent.tsx

```diff
@@ -0,0 +1,88 @@
+import { Button, Card, Flex, Typography } from 'antd';
+import { entries } from 'lodash';
+import Image from 'next/image';
+import { useCallback } from 'react';
+
+import { AGENT_CONFIG } from '@/config/agents';
+import { COLOR } from '@/constants/colors';
+import { AgentType } from '@/enums/Agent';
+import { Pages } from '@/enums/Pages';
+import { SetupScreen } from '@/enums/SetupScreen';
+import { usePageState } from '@/hooks/usePageState';
+import { useServices } from '@/hooks/useServices';
+import { AgentConfig } from '@/types/Agent';
+
+import { CardFlex } from '../styled/CardFlex';
+import { SetupCreateHeader } from './Create/SetupCreateHeader';
+
+const { Title, Text } = Typography;
+
+type EachAgentProps = { agentType: AgentType; agentConfig: AgentConfig };
+const EachAgent = ({ agentType, agentConfig }: EachAgentProps) => {
+  const { selectedAgentType, updateAgentType } = useServices();
+  const { goto } = usePageState();
+
+  const isCurrentAgent = selectedAgentType === agentType;
+
+  const handleSelectAgent = useCallback(() => {
+    updateAgentType(agentType);
+    goto(Pages.SetupYourAgent); // TODO: page to be added
```

Comment:
can we support dynamic page here? can be either onboarding part or main, but should somehow pass it